### PR TITLE
Split `LICENSE` file

### PR DESCRIPTION
**Pull request by not522**
## Motivation

The current `LICENSE` file contains the licenses of Optuna and third-pary libraries. This prevents the automatic license recognition of GitHub.

## Description of the changes

This PR splits the license file to `LICENSE` and `LICENSE_THIRD_PARTY`. It also updates `MANIFEST.in` to keep the copyright notice of the third-party libraries on distributions. Please note that it does not affect wheel distribution, however files with the name `LICEN[CS]E*` are automatically included in the distribution (See https://wheel.readthedocs.io/en/stable/user_guide.html#including-license-files-in-the-generated-wheel-file).

I have confirmed that the distribution generated by `python -m build` includes the copyright notice.

The purpose of updating `README.md` is to make it easier for users who incorporate Optuna code as is to be aware of the license information of third-party libraries.

### CODE REVIEW

**Comment by not522** [comment]
@c-bata @HideakiImamura Could you review this PR?

**Review by HideakiImamura — COMMENTED** [review]
Does the current change ensure that the GitHub recognize Optuna is under MIT licence?

**Comment by not522** [comment]
Yes. I checked it on my fork repository. https://github.com/not522/optuna (Temporarily, the default branch of this repository includes the change of this PR.)

File: LICENSE_THIRD_PARTY [hunk 1]
```diff
@@ -0,0 +1,52 @@
+Optuna contains code that is licensed by third-party developers.
```
Comment (c-bata) [inline LICENSE_THIRD_PARTY:1]: Do you consider using  `license_files` option, which is supported from setuptools `42.0.0`?

* https://setuptools.pypa.io/en/latest/userguide/declarative_config.html#metadata
* https://github.com/scipy/scipy/blob/550695995a48c5707d962367a07e6295dc8b307c/pyproject.toml#L37-L39

**Review by c-bata — COMMENTED** [review]
Thank you for your pull request. I confirmed that sdist package includes `LICENSE_THIRD_PARTY` file.

Changes almost look good to me. I left one question though.

File: LICENSE_THIRD_PARTY (hunk 1 above)
Comment (not522) [reply LICENSE_THIRD_PARTY:1]: Thank you. In my understanding, `license-files` can be specified in `tool.setuptools` of `pyproject.toml`. But the default value is enough to cover our license files. https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration

Comment (c-bata) [reply LICENSE_THIRD_PARTY:1]: Thank you for letting me know. Your understanding seems to be correct :bow:

```
$ python
Python 3.11.7 (main, Jan 20 2024, 14:32:50) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from importlib import metadata
>>> for dist in metadata.distributions():
...     if dist.metadata['Name'] == "optuna":
...         optuna_dist = dist
... 
>>> optuna_dist.metadata.get_all("License-File")
['LICENSE', 'LICENSE_THIRD_PARTY']
```

**Review by c-bata — APPROVED** [review]
LGTM!

**Review by HideakiImamura — APPROVED** [review]

**Pull request merged**